srcs/server/server.cpp

```diff
@@ -76,21 +75,25 @@ namespace ft
 		}
 	}	
 
-	bool Server::recieve_request_()
+	void Server::recieve_request_()
 	{
 		Socket::RecievedMsg recieved_msg;
 		unsigned int		response_code;
 		std::string			response;
 
 		try
-		{	
+		{
 			remove_timeout_clients_();
+
 			recieved_msg = socket_.recieve_msg();
 
 			print_debug_(recieved_msg);
-	
-			ServerChild& serverChild = httpRequest_pair_map_[recieved_msg.client_id].second;
 
+			// return from this loop if cgi message has been sent
+			if (handle_cgi(recieved_msg))
+				return ;
```

Comment:
handle_cgi will only happen in the case that the recieved_msg fd is the cgi fd that is set up.
it is set up from the throw statement in http_method do_cgi.  
that exception readCGI will be caught below. it adds cgi fd to the socket fd_poll_vec. 
Then after poll we will catch the cgi_fd in the POLLHUP if statement (because the pipe is closed)
the socket will read the cgi_fd, get the contents and return them in recieved_msg which will be caught here.
From here handle_cgi will send the output of cgi_fd to the original client who requested the transaction